docs/testplan/GCU-Dynamic-ACL-testplan.md

```diff
@@ -0,0 +1,471 @@
+
+#### Test Objective
+
+Verify that we can remove a previously created drop rule from our ACL Table with GCU
+
+#### Testing Steps
+
+- Use GCU to create a drop rule on a specific port on ACL Table
+
+- Remove the drop rule from ACL Table
+
+- Verify that all operations were successful
+
+- Verify that the result of "show acl rule {rule_name}" has no relevant output
+
+- Verify that packets that were previously dropped are now forwarded
+
+### Test Case # 4 - Create forward rules within custom ACL table
+
+#### Test Objective
+
+Verify that we can create a forward rule utilizing GCU, and that we can create forward rules for both ipv4 and ipv6
+
+#### Testing Steps
+
+- Use GCU to create 2 new forwarding rules with top priority on our ACL Table, one for IPv4 and one for IPv6
+
+- Use GCU to create drop rule on a specific port with lower priority in ACL Table
+
+- Verify that all operations were successful
+
+- Verify that for both rules created, "show acl rule | grep {rulename}" matches expected output for both rules
+
+- Verify that packets matching forwarding rules on this specific port are correctly forwarded
+
+- Verify that packets not matching forwarding rules on this specific port are correctly dropped
+
+### Test Case # 5 - Replace the IP Address on an ACL Rule
+
+#### Test Objective
+
+Verify that after creation, ACL Rules can have their match conditions updated
+
+#### Testing Steps
+
+- Use GCU to create 2 new forwarding rules on ACL Table
+
+- Use GCU to create drop rule on a specific port with lower priority on ACL Table
+
+- Use GCU to replace the IP addresses in both forwarding rules
+
+- Verify that all operations were successful
+
+- Verify that the results of "show acl rule | grep {rule_name}" matches expected output for both rules
+
+- Verify that packets with IPs matching original forwarding rules on this specific port are dropped
+
+- Verify that packets with IPs matching replacement rules on this specific port are forwarded
+
+### Test Case # 6 - Remove forward rule from ACL Table
+
+#### Test Objective
+
+Verify that after creation, a forward ACL rule can be removed and packets matching the forward rule are no longer forwarded
+
+#### Testing Steps
+
+- Use GCU to create 2 new forwarding rules on ACL Table
+
+- Use GCU to create drop rule on a specific port with lower priority on ACL Table
+
+- Use GCU to remove the 2 forwarding rules
+
+- Verify that all operations were successful
+
+- Verify that the results of "show acl rule {rule_name}" are empty for both rule names
+
+- Verify that packets with IPs matching the removed forwarding rules are dropped on this specific port
+
+### Test Case # 7 - Scale test of ACL Table, add large amount of forward and drop rules
+
+#### Test Objective
+
+Verify that GCU is capable of adding a large amount of ACL rules and that they are still followed.
+
+#### Testing Steps
+
+- Use GCU to create ~100 forwarding rules on ACL Table
+
+- Use GCU to create ~20 drop rules on ACL Table
+
```

Comment:
Please check what's the maximum number of the forwarding rule (ipv4 + ipv6). I roughly remember the number should be above 100. And for the dropping rules, the maximum number should equal to the port number, i.e. all server facing ports are isolated.